src/MPAGSCipher/ProcessCmdLine.cpp

```diff
@@ -0,0 +1,162 @@
+#include <iostream>
+#include <vector>
+#include <string>
+
+#include "ProcessCmdLine.hpp"
+
+bool processCmdLine(const std::vector<std::string>& args,
+                    bool& helpRequested,
+                    bool& versionRequested,
+                    std::string& inputFileName,
+                    std::string& outputFileName,
+                    bool& encrypt,
+                    bool& decrypt,
+                    bool& keySupplied,
+                    int& key)
```

Comment:
These 4 options can be covered with just 2

`encrypt` - `true` => encrypt, `false` => decrypt
`key` = Stores the key

Note that the key should also be unsigend.